nens_auth_client/backends.py

```diff
@@ -112,3 +112,47 @@ def authenticate(self, request, claims):
         create_remote_user(user, claims)
 
         return user if self.user_can_authenticate(user) else None
+
+
+class AcceptNensBackend(ModelBackend):
+    def authenticate(self, request, claims):
+        """Backend for auto-accepting users that have a N&S azure AD account.
+
+        The behaviour looks a bit like the SSOMigrationBackend above, but with
+        two key differences:
+
+        - N&S users don't need an existing account, they're accepted right
+          away and a User is created if missing.
+
+        - We only deal with users recognized as being really from N&S.
+
+        Args:
+          request: the current request
+          claims (dict): the verified payload of the ID or Access token
+
+        Returns:
+          user or None
+
+        """
+        username = _nens_user_extract_username(claims)
+        if username is None:
+            return
+
+        email = claims.get("email")
+
+        try:
+            user, created = UserModel.objects.get_or_create(
+                username__iexact=username, email__iexact=email
+            )
+        except MultipleObjectsReturned:
+            raise PermissionDenied(settings.NENS_AUTH_ERROR_USER_MULTIPLE)
+        if created:
+            logger.info("Auto-accepting new N&S user %s: created", username)
+
+        if not self.user_can_authenticate(user):
+            raise PermissionDenied(settings.NENS_AUTH_ERROR_USER_INACTIVE)
+
+        # Create a permanent association
+        create_remote_user(user, claims)
+
+        return user if self.user_can_authenticate(user) else None
```

Comment:
`user_can_authenticate` wordt nu voor de 2e keer aangroepen (zat ook al in de `SSOMigrationBackend` zie ik)

Reply:
Zappppppp. Goeie.